flowbased-computation/flowbased-computation-impl/src/main/java/com/farao_community/farao/flowbased_computation/impl/LoopFlowUtil.java

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * Loop flow util is used
+ * - for loop flow calculation (used in search tree rao for CCR Core), and
+ * - for Min RAM adjustment calculation
+ * @author Pengbo Wang {@literal <pengbo.wang at rte-international.com>}
+ */
+public final class LoopFlowUtil {
+    private LoopFlowUtil() {
+    }
+
+    /**
+     * calculate loop flow
+     * input CracFile
+     * This function could be removed when Flowbased computation no longer use CracFile but Crac
+     */
+    public static Map<String, Double> calculateLoopFlows(Network network,
+                                                         CracFile cracFile,
+                                                         GlskProvider glskProvider,
+                                                         List<String> countries,
+                                                         ComputationManager computationManager,
+                                                         FlowBasedComputationParameters flowBasedComputationParameters
+    ) {
+        //Call flowbased computation on given glskProvider
+        FlowBasedComputationProvider flowBasedComputationProvider = new FlowBasedComputationImpl();
+        FlowBasedComputationResult flowBasedComputationResult = flowBasedComputationProvider.run(network,
+                cracFile, glskProvider, computationManager, network.getVariantManager().getWorkingVariantId(), flowBasedComputationParameters)
+                .join();
+
+        Map<String, Double> frefResults = frefResultById(flowBasedComputationResult); //get reference flow
+        Map<String, Map<String, Double>> ptdfResults = ptdfResultById(flowBasedComputationResult); // get ptdf
+        Map<String, Double> referenceNetPositionByCountry = getRefNetPositionByCountry(network, countries); // get Net positions
+        return buildLoopFlowsFromResult(flowBasedComputationResult, frefResults, ptdfResults, referenceNetPositionByCountry);
+    }
+
+    /**
+     * calculate loop flow
+     * input Crac
+     */
+    public static Map<String, Double> calculateLoopFlows(Network network,
+                                                         Crac crac,
+                                                         GlskProvider glskProvider,
+                                                         List<String> countries,
+                                                         ComputationManager computationManager,
+                                                         FlowBasedComputationParameters flowBasedComputationParameters
+    ) {
+        //Call flowbased computation on given glskProvider
+        FlowBasedComputationCracImpl flowBasedComputationProvider = new FlowBasedComputationCracImpl();
+        FlowBasedComputationResult flowBasedComputationResult = flowBasedComputationProvider.run(network,
+                crac, glskProvider, computationManager, network.getVariantManager().getWorkingVariantId(), flowBasedComputationParameters)
+                .join();
```

Comment:
This should ideally be done using the FlowBasedComputation api instead of using a hard-coded FlowBased computation implementation.

FlowBasedComputation.run(....)

Here we are however stuck by the fact that one impl uses the cracFile, while one other uses a Crac through a method not signed in the API.

This might be a temporary fix to be able to make flow-based computations without duplicating the api. But we need to trace this unconventional use of the provider.

(see all the todo)

Reply:
added todo